bank.py

```diff
@@ -21,3 +22,27 @@ def get_companies(self, name):
     @classmethod
     def search_companies(cls, name, clause):
         return [('bank.party.companies',) + tuple(clause[1:])]
+
+
+class BankAccountNumber(metaclass=PoolMeta):
+    __name__ = 'bank.account.number'
+
+    @classmethod
+    def __register__(cls, module_name):
+        pool = Pool()
+        Module = pool.get('ir.module')
+
+        cursor = Transaction().connection.cursor()
+
+        party_companies = Module.search([
+            ('name', '=', 'party_company'),
+            ('state', '!=', 'activated'),
+            ], limit=1)
+        if party_companies:
+            cursor.execute("ALTER TABLE bank_account_number DROP CONSTRAINT "
+                "IF EXISTS bank_account_number_number_iban_exclude")
+            cursor.execute("ALTER table bank_account_number add constraint "
+                "bank_account_number_number_iban_exclude check (type in "
+                "('iban', 'other'))")
```

Comment:
I think using type != 'x' would make it more robust in case in the future other types are added.